tests/pstl-tests/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+function(compile_test source)
+    set(test_name "pstl.${source}")
+    set(source "${source}.cpp")
+    add_executable(${test_name} ${source})
+    target_link_libraries(${test_name} PUBLIC ${gtest_BINARY_DIR}/libgtest.a
+                                       PUBLIC ${gtest_BINARY_DIR}/libgtest_main.a
+                                       PUBLIC pthread)
+    add_dependencies(${test_name} gtest_main)
+    add_dependencies(${test_name} gtest)
+    add_sycl_to_target(${test_name} ${CMAKE_CURRENT_BINARY_DIR}
+                                    ${CMAKE_CURRENT_SOURCE_DIR}/${source})
+    add_test(test.${test_name} ${test_name})
+endfunction(add_test)

```

Comment:
the function is called `compile_test`. I would just write `endfunction()` instead of naming the function twice.

Reply:
I thought this was illegal. Glad it's not :smiley:

Reply:
Fixed.